autofill/autofill-store/src/main/java/com/duckduckgo/autofill/store/SecureStoreBackedAutofillStore.kt

```diff
@@ -0,0 +1,191 @@
+
+        matchingCredentials.forEach {
+            val modifiedDetails = it.details.copy(username = credentials.username)
+            val modified = it.copy(password = credentials.password, details = modifiedDetails)
+            secureStorage.updateWebsiteLoginDetailsWithCredentials(modified)
+        }
+    }
+
+    override suspend fun getAllCredentials(): Flow<List<LoginCredentials>> {
+        return secureStorage.websiteLoginDetailsWithCredentials()
+            .map { list ->
+                list.map { it.toLoginCredentials() }
+            }
+    }
+
+    override suspend fun deleteCredentials(id: Int) {
+        secureStorage.deleteWebsiteLoginDetailsWithCredentials(id)
+    }
+
+    override suspend fun updateCredentials(credentials: LoginCredentials) {
+        secureStorage.updateWebsiteLoginDetailsWithCredentials(credentials.toWebsiteLoginCredentials())
+    }
+
+    override suspend fun containsCredentials(
+        rawUrl: String,
+        username: String,
+        password: String
+    ): ContainsCredentialsResult {
+        val url = rawUrl.extractSchemeAndDomain() ?: return NoMatch
+        val credentials = secureStorage.websiteLoginDetailsWithCredentialsForDomain(url).firstOrNull() ?: return NoMatch
+
+        var exactMatchFound = false
+        var usernameMatchFound = false
+        var urlMatch = false
+
+        credentials.forEach {
+            urlMatch = true
+
+            if (it.details.username == username) {
+                usernameMatchFound = true
+                if (it.password == password) {
+                    exactMatchFound = true
+                }
+            }
+        }
+
+        return if (exactMatchFound) {
+            ContainsCredentialsResult.ExactMatch
+        } else if (usernameMatchFound) {
+            ContainsCredentialsResult.UsernameMatch
+        } else if (urlMatch) {
+            ContainsCredentialsResult.UrlOnlyMatch
+        } else {
+            NoMatch
+        }
+    }
+
+    private fun WebsiteLoginDetailsWithCredentials.toLoginCredentials(): LoginCredentials {
+        return LoginCredentials(
+            id = details.id,
+            domain = details.domain,
+            username = details.username,
+            password = password
+        )
+    }
+
+    private fun LoginCredentials.toWebsiteLoginCredentials(): WebsiteLoginDetailsWithCredentials {
+        return WebsiteLoginDetailsWithCredentials(
+            details = WebsiteLoginDetails(domain = domain, username = username, id = id),
+            password = password
+        )
+    }
+
+    companion object {
+        const val FILENAME = "com.duckduckgo.autofill.store.autofill_store"
+        const val AUTOFILL_ENABLED = "autofill_enabled"
+    }
+}
+
+@Module
+@ContributesTo(AppScope::class)
+class AutofillStoreModule {
+
+    @Provides
+    @SingleInstanceIn(AppScope::class)
+    fun autofillStore(
```

Comment:
NIT: from discussion today, we can move this dagger module into impl instead of here. But as mentioned, it's jusr the way we already do things now but it's not a big blocker to not do it.